gcp-resources/build.gradle.kts

```diff
@@ -15,7 +15,7 @@ dependencies {
   implementation("com.google.cloud.opentelemetry:detector-resources-support:0.28.0")
 
   implementation("io.opentelemetry.semconv:opentelemetry-semconv")
-  implementation("io.opentelemetry.semconv:opentelemetry-semconv-incubating:1.25.0-alpha")
+  implementation("io.opentelemetry.semconv:opentelemetry-semconv-incubating")
```

Comment:
All these usages of `opentelemetry-semconv-incubating` go against our [stated advice](https://github.com/open-telemetry/semantic-conventions-java?tab=readme-ov-file#published-released):

> Library instrumentation SHOULD NOT depend on this.

> Libraries can use this for testing, but should make copies of the attributes to avoid possible runtime errors from version conflicts.

I think we should follow our own advice to avoid unresolvable transitive version conflicts.

Reply:
Thanks for the reminder, I completely missed it.

I'm definitely missing the context on the intent of this advice, so maybe there is something to clarify here.

For now, the way I understand things:

The `opentelemetry-semconv` artifact only contains stable semantic conventions.
The `opentelemetry-semconv-incubating` artifact only contains incubating semantic conventions.

Consumers of semantic conventions who need/want to avoid breaking change:

- should have a direct dependency on `opentelemetry-semconv`.
- might have a test dependency on `opentelemetry-semconv-incubating`.
- should inline the incubating semconv attributes in the production code, this is where they trade maintainability for stability.
  
Consumers of semantic conventions who can tolerate/handle breaking changes:

- should have a direct dependency on `opentelemetry-semconv`
- should have a direct dependency on `opentelemetry-semconv-incubating` which provides an easy way to find usages and refactor in the IDE.

For example:
- in the instrumentation agent, we already use the incubating artifact and can embrace breaking changes quite easily due to the large set of contributors who can help. However, the goal is to have as much stable as possible and sometime the incubating metrics (like the JVM metrics) are opt-in through configuration, however this option is not always possible at attribute level.
- for a module in contrib that has only two contributors and is rarely updated, keeping it stable is probably a better option, so having some duplication is a good compromise.

To me, by adding a dependency on `opentelemetry-semconv-incubating` on a module in contrib, you opt-in for some breaking changes.

Maybe here having a single version used for all the contrib modules is a bold move and having each module handle its own version (even if they are currently aligned) is a better compromise, so each module can be updated at its own pace.